### PR TITLE
fix: @apply darkmode in components

### LINKED ISSUE #54: Dark mode does not get applied in component styles
## I'm submitting a...

<pre><code>[x] Bug report  
</code></pre>

## Current behavior

Dark mode stops working when extracting class into the component css using `@apply`

## Expected behavior

Dark mode should keep working.

## Minimal reproduction of the problem with instructions

- Create an app with dark mode enabled
- Add styles prefixed with `dark:`
- Verify they work as expected
- Extract out the classes into the component styles, using `@apply`
- Verify it no longer works as expected.

I got a demo in my [video tutorial on YouTube](https://www.youtube.com/watch?t=93&v=XFxuKUBUqao&feature=youtu.be). You can see that after applying the styles, the dark mode disappears.

## Environment

<pre><code>
Angular version: 11.0.x

Browser:
- [x] Brave version V1.18.78
 
</code></pre>

### PR DESCRIPTION
**Pull request by vltansky**
closes #54

### CODE REVIEW

**Comment by nartc** [comment]
can we just install this plugin when darkMode: 'class' is selected?

**Comment by vltansky** [comment]
PostCSS plugin is 25 lines. I don't think there is a point to conditional add it.
Maybe we should just insert the plugin inside ngneat/tailwind? Or you think it should be separated?

**Comment by nartc** [comment]
@vltansky that is totally up to you. Do you want to include this additional plugin inside of ngneat/tailwind?

**Comment by vltansky** [comment]
Yeah, there is no use to it outside of ngneat/tailwind

**Comment by vltansky** [comment]
But not sure how to test it. Right now I'm testing postcss with jest.

**Comment by vltansky** [comment]
you can use this starter to test it: https://github.com/vltansky/tailwind-test2
run `ng add @vltansky/tailwind`.

plugin: https://github.com/vltansky/postcss-ng-tailwind-in-components

File: libs/tailwind/src/constants/dependencies.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export const DEPENDENCIES = [
   'tailwindcss',
   '@angular-builders/custom-webpack',
+  'postcss-ng-tailwind-in-components',
```
Comment (nartc) [inline libs/tailwind/src/constants/dependencies.ts:4]: suggestion (non-blocking): can we make this `postcss-ng-tailwind-in-components` installed only when `darkMode: class` is selected from the schematics?

**Review by nartc — APPROVED** [review]

**Comment by nartc** [comment]
Approved the PR with 1 non-blocking suggestion. We can discuss and follow-up the suggestion in a separate issue/PR if you'd like.

**Pull request merged**